File: works/2526.cpp
#include <iostream>
#include <unordered_map>
using namespace std;

int main()
{
    int a, b; cin >> a >> b;
    unordered_map<int,int> m;
    int num = a;
    int cnt = 1;
    while(true)
    {
        // 이미 출현한 수 나오면 모든 수 갯수에서 처음으로 중복 출현한 수 이전 수의 갯수를 뺴면 싸이클의 갯수
        if(m[num] > 0)
        {
            cout << m.size() - (m[num]-1);
            break;
        }
        m[num] = cnt++;
        num = num * a % b;
    }
}

File: works/js/4-1/2526.js
let fs = require('fs');
let input = fs.readFileSync(0).toString().trim().split('\n')
// let input = fs.readFileSync('input.txt').toString().split('\n')

let [a,b] = input[0].split(' ').map(Number) 
let m = new Map() 
let num = a, cnt = 1

while(true) {
    if(m.has(num)) {
        console.log(m.size - (m.get(num) - 1))
        break 
    }
    m.set(num, cnt++)
    num = num * a % b 
}
